pkg/util/kubernetes/kubelet/podwatcher.go

```diff
@@ -122,3 +147,24 @@ func (w *PodWatcher) Expire() ([]string, error) {
 func (w *PodWatcher) GetPodForEntityID(entityID string) (*Pod, error) {
 	return w.kubeUtil.GetPodForEntityID(entityID)
 }
+
+// digestMap return a unique hash for a map
+// used to track changes in labels and annotations values
+func digestMap(m map[string]string) string {
+	if m == nil {
```

Comment:
I think empty maps aren't a special case to handle (will not break the function logic or cause a runtime error) the function will generate a unique hash to an empty map, WDYT?

Reply:
I believe it would be nice to just change the nil test to `if len(m) == 0 {` and return an empty string for both cases so we don't compute an additional hash for that.

Reply:
fair enough 👍